make_mock_json.py

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Copyright (c) 2019 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+import argparse
+import copy
+import logging
+import os
+import sys
+import json
+import subprocess
+from textwrap import dedent
+
+from osbs import set_logging
+from osbs.api import OSBS
+from osbs.conf import Configuration
+from osbs.constants import DEFAULT_CONFIGURATION_FILE, DEFAULT_ARRANGEMENT_VERSION
+from tests.constants import TEST_BUILD, TEST_ORCHESTRATOR_BUILD, TEST_CANCELLED_BUILD
+from osbs.exceptions import OsbsException, OsbsResponseException
+from osbs.cli.capture import setup_json_capture
+
+
+OS_VERSION = "3.9.41"
+ORCH_BUILD_LOG = dedent("""2017-06-23 17:18:41,791 platform:- - atomic_reactor.foo - DEBUG - this is from the orchestrator build
+2017-06-23 17:18:41,791 platform:x86_64 - atomic_reactor.foo - INFO - 2017-06-23 17:18:41,400 platform:- atomic_reactor.foo -  DEBUG - this is from a worker build
+2017-06-23 05:04:51,334 platform:x86_64 - atomic_reactor.plugins.orchestrate_build - INFO - "ContainersPaused": 0,
+2017-06-23 17:18:41,791 - I really like bacon
+""")
+BASE_BUILD_LOG = u"   líne 1   \n".encode('utf-8')
+MOCK_CONFIG_MAP = {
+    "apiVersion": "v1",
+    "data": {
+        "special.how": "\"very\"",
+        "special.type": "{\"quark\": \"charm\"}",
+        "config.yaml": "version:1",
+        "config.yml": "version:2",
+        "config.ymlll": "{\"version\":3}",
+        "config.json": "{\"version\":4}"
+    },
+    "kind": "ConfigMap",
+    "metadata": {
+        "creationTimestamp": "2017-06-14T16:49:47Z",
+        "name": "special-config",
+        "namespace": "osbs-qa01",
+        "resourceVersion": "117005745",
+        "selfLink": "/api/v1/namespaces/osbs-qa01/configmaps/special-config",
+        "uid": "797c8350-5121-11e7-9ce0-0efeb5d2209c"
+    }
+}
+DEFAULT_DIR = "tests/mock_jsons"
+
+
+def canonize_data(build_data, build_name=TEST_BUILD, build_status=None):
+    build_data["status"]["config"] = {
+        "kind": "BuildConfig",
+        "name": build_name,
+        "namespace": "default",
+    }
+    build_data["metadata"]["name"] = build_name
+    if build_status:
+        build_data["status"]["phase"] = build_status
+        try:
+            del build_data["status"]["logSnippet"]
+            del build_data["status"]["message"]
+        except KeyError:
+            pass
+    return build_data
+
+
+def find_or_make_dir(dir_name):
```

Comment:
Fine for Python 2. In Python 3 we'll be able to `os.makedirs(..., exist_ok=True)`.

Reply:
Fixed.